service/core/client/src/main/scala/com/lightbend/lagom/internal/client/WebSocketClient.scala

```diff
@@ -42,12 +44,12 @@ import scala.collection.immutable
 /**
  * A WebSocket client
  */
-private[lagom] abstract class WebSocketClient(environment: Environment, config: Config, eventLoop: EventLoopGroup,
+private[lagom] abstract class WebSocketClient(environment: Environment, wsClientConfig: WebSocketClientConfig, eventLoop: EventLoopGroup,
                                               lifecycle: ApplicationLifecycle)(implicit ec: ExecutionContext) extends LagomServiceApiBridge {
 
   lifecycle.addStopHook(() => shutdown())
 
-  val maxFrameLength = config.getBytes("lagom.client.websocket.frame.maxLength").toInt
+  val maxFrameLength = math.min(Int.MaxValue.toLong, wsClientConfig.config.getBytes("frame.maxLength")).toInt
```

Comment:
I also added an overflow check as the parameter provided to the Netty handshaker must be an Int (and not a Long)